### ISSUE TITLE
preloadjs and soundjs in different bundles

### ISSUE THREAD

**Issue by robotlabs** [CreateJS/SoundJS#224]
Hey there, how can i play some audio preloaded with preloadjs where the soundjs is in a different bundle, preloaded as well with preloadjs. So preloadjs is in one bundle, and soundjs is in another one preloaded later. it looks it is working only if i preload ( again ) the sounds.

**Comment by lannymcnie** [comment]
Loading sounds with PreloadJS (to be used with SoundJS) requires the sounds that are loaded to be "registered" with SoundJS so they can be played.

**Comment by robotlabs** [comment]
yes, i know. but if i preload the sounds, without having the soundjs in place, and then i attach soundJS lib, and i register sounds, the sounds are being downloaded again. that makes preloadjs useless. is there anyway to tell soundJS to get the audio preloaded by preloadJS instead of re-loading again ? my problem is that adding soundJs with preloadJS just make the size of the preload 60 Kb bigger

**Comment by gskinner** [comment]
This may be a silly question, but if the initial load size is a major issue, why are you preloading audio that you aren't going to play? It would seem to make sense to simply load your audio when you load SoundJS.

**Comment by lannymcnie** [comment]
Currently there is no way to register sounds that are already preloaded. Preloading for webaudio requires XHR loading, which doesn't exactly live in the cache -- and there is no way to associate a loaded context with SoundJS. The other issue is that SoundJS uses a single audio context for playback.

We have plans to support multiple contexts, and hopefully we can add support for registering preloaded audio, but for now it is not possible.

**Comment by robotlabs** [comment]
ok thanks a lot

**Comment by gskinner** [comment]
I'd still be interested in the use case though, so we can evaluate priority level for this feature.

**Comment by robotlabs** [comment]
Hey Grant. Sorry, haven't  seen your question when i closed . Let me explain my scenario.
I have 2 bundles for my app. loader.js and app.js
loader.js is the only js included in the html. 
loader.js ( that contains preloadJS ) loads the main bundle ( app.js ) and all the assets, like audio, images.
I need the sounds to be ready and can't  preload later. the experience need to be seamless after the preload.
So my problem with this is that i have to include soundJS in the loader, in order to be able to play the sounds, and this increase my loader.js size of ish 60KB. I was trying to avoid that, and wanted to have soundJS included in app.js.

**Comment by lannymcnie** [comment]
One issue that would need to be solved is that PreloadJS has no logic for loading web audio (just HTML5 tag-based audio). In order to use PreloadJS this way, you would have to specify the load type as [binary](http://createjs.com/docs/preloadjs/classes/AbstractLoader.html#property_BINARY), with a custom `responseType` so it loads properly. You would also have to determine the proper format (mp3 vs ogg, etc), and only then could it be passed into SoundJS. 

We will definitely look at ways to support this, as this issue has come up a few times, but you can see how its not quite as simple as registering the loaded audio.

**Comment by robotlabs** [comment]
ye all make sense. Cool. If I have time will look at binary approach and if any good results will let you know guys. thanks a lot
